creator-node/src/index.ts

```diff
@@ -127,14 +99,84 @@ const startApp = async () => {
     logger.warn(`Could not clear write locks. Skipping..: ${e.message}`)
   }
 
-  // Initialize services that do not require the server, but do not need to be awaited.
-  serviceRegistry.initServicesAsynchronously()
+  const numWorkers = clusterUtils.getNumWorkers()
+  logger.info(`Spawning ${numWorkers} processes to run the Express app...`)
+  const firstWorker = cluster.fork()
+  // Wait for the first worker to perform one-time init logic before spawning other workers
+  firstWorker.on('message', (msg) => {
+    if (msg?.cmd === 'initComplete') {
+      for (let i = 0; i < numWorkers - 1; i++) {
+        cluster.fork()
+      }
+    }
+  })
 
-  // Some Services cannot start until server is up. Start them now
-  // No need to await on this as this process can take a while and can run in the background
-  serviceRegistry.initServicesThatRequireServer(appInfo.app)
+  for (const worker of Object.values(cluster.workers || {})) {
+    worker?.on('message', (msg) => {
+      if (msg?.cmd === 'setSpecialWorkerId') {
+        clusterUtils.specialWorkerId = msg?.val
+      }
+    })
+  }
 
-  // when app terminates, close down any open DB connections gracefully
+  cluster.on('exit', (worker, code, signal) => {
+    logger.info(
+      `Worker process with pid=${worker.process.pid} died because ${
+        signal || code
+      }. Respawning...`
+    )
+    const newWorker = cluster.fork()
+    if (clusterUtils.specialWorkerId === worker.id) {
+      logger.info(
+        'The worker that died was the special worker. Setting a new special worker...'
+      )
+      clusterUtils.specialWorkerId = newWorker.id
```

Comment:
we only change the special worker when it dies. this is a little tricky but the point here is to change state on the new special process so that it knows it's special before it inits so when it inits it registers the onComplete listeners for state machine

Reply:
yep makes sense after our discussion